### PR TITLE
Prevent crash when sending `nil` site ID to ReaderHelpers.dispatchToggleFollowSiteMessage

### PR DESCRIPTION
**Pull request by twstokes**
## Description

This is an attempt to prevent the crash in https://github.com/wordpress-mobile/WordPress-iOS/issues/21307, but does not resolve the root issue. A new Sentry error is sent in place of the (hopefully prevented) crash.

Related: https://github.com/wordpress-mobile/WordPress-iOS/pull/21140

## Testing

Follow and unfollow sites in Reader in the following areas and expect the app to not crash:

| Area 1 | Area 2 | Area 3 |
| - | - | - |
| IMG_0438 | IMG_0439 | IMG_0440 |

Some different ways to reach these areas:
- Tapping posts in Reader Discover or search results
- Tapping site titles in Reader Discover or search results

## Regression Notes
1. Potential unintended areas of impact
- None identified

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manually tested the steps above

3. What automated tests I added (or what prevented me from doing so)
- None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Models/ReaderSiteTopic+Lookup.swift [hunk 1]
```diff
@@ -16,8 +16,12 @@ extension ReaderSiteTopic {
     ///
     /// - Parameter siteID: The site id of the topic
     @objc(lookupWithSiteID:inContext:)
-    static func objc_lookup(withSiteID siteID: NSNumber, in context: NSManagedObjectContext) -> ReaderSiteTopic? {
-        try? lookup(withSiteID: siteID, in: context)
+    static func objc_lookup(withSiteID siteID: NSNumber?, in context: NSManagedObjectContext) -> ReaderSiteTopic? {
```
Comment (twstokes) [inline WordPress/Classes/Models/ReaderSiteTopic+Lookup.swift:19]: A similar treatment as https://github.com/wordpress-mobile/WordPress-iOS/pull/21140.

File: WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift [hunk 2]
```diff
@@ -366,18 +366,23 @@ struct ReaderPostMenuButtonTitles {
...
+            CrashLogging.main.logMessage("Expected siteID to exist", level: .error)
+            return
+        }
+
         let blogName = {
             guard let blogNameForDisplay = post.blogNameForDisplay() else {
-                if let siteID = post.siteID, let postID = post.postID {
-                    CrashLogging.main.logMessage("Expected blogNameForDisplay() to exist",
-                                                 properties: ["siteID": siteID, "postID": postID],
+                CrashLogging.main.logMessage("Expected blogNameForDisplay() to exist",
```
Comment (twstokes) [inline WordPress/Classes/ViewRelated/Reader/ReaderHelpers.swift:378]: I wanted to make this a little less restrictive.

**Review by staskus — APPROVED** [review]
Thanks, LGTM! ✅

**Review by salimbraksa — APPROVED** [review]
Thanks for the fix @twstokes. LGTM! 🚀

**Pull request merged**